manticore/native/models.py

```diff
@@ -140,3 +142,113 @@ def strlen(state, s):
             ret = ITEBV(cpu.address_bit_size, byt == 0, offset, ret)
 
     return ret
+
+
+def is_NULL(byte, constrs) -> bool:
+    """
+    Checks if a given byte read from memory is NULL or effectively NULL
+
+    :param byte: byte read from memory to be examined
+    :param constrs: state constraints
+    :return: whether a given byte is NULL or constrained to NULL
+    """
+    if issymbolic(byte):
+        return not Z3Solver.instance().can_be_true(constrs, byte != 0)
+    else:
+        return byte == 0
+
+
+def cant_be_NULL(byte, constrs) -> bool:
+    """
+    Checks if a given byte read from memory is not NULL or cannot be NULL
+
+    :param byte: byte read from memory to be examined
+    :param constrs: state constraints
+    :return: whether a given byte is not NULL or cannot be NULL
+    """
+    if issymbolic(byte):
+        return not Z3Solver.instance().can_be_true(constrs, byte == 0)
+    else:
+        return byte != 0
```

Comment:
Are these functions the inverse of each other?

Is it true that `not is_NULL()` is the same as `cant_be_NULL`?

Reply:
Forgot about the `can_be_NULL`... Let me take a closer look

Reply:
Actually, these do make sense to me now 👍